components/ILIAS/Forum/classes/Moderation/class.ForumModeratorsTable.php

```diff
@@ -0,0 +1,198 @@
+    protected \ilLanguage $lng;
+    protected UI\Factory $ui_factory;
+    protected ServerRequestInterface $request;
+    protected Data\Factory $data_factory;
+
+    public function __construct(ilForumModerators $forum_moderators)
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->lng = $DIC->language();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->request = $DIC->http()->request();
+        $this->data_factory = new Data\Factory();
+        $this->forum_moderators = $forum_moderators;
+    }
+
+    public function getComponent(): UI\Component\Table\Data
+    {
+        $columns = $this->getColumns();
+        $actions = $this->getActions();
+        $data_retrieval = $this->getDataRetrieval();
+
+        return $this->ui_factory->table()
+                                ->data($this->lng->txt('frm_moderators'), $columns, $data_retrieval)
+                                ->withActions($actions)
+                                ->withRequest($this->request);
+    }
+
+    protected function getColumns(): array
+    {
+        return [
+            'usr_id' => $this->ui_factory->table()->column()->number('User ID')
+                                         ->withIsSortable(true),
+
+            'login' => $this->ui_factory->table()->column()->text($this->lng->txt('login'))
+                                        ->withIsSortable(true),
+
+            'firstname' => $this->ui_factory->table()->column()->text($this->lng->txt('firstname'))
+                                            ->withIsSortable(true),
+
+            'lastname' => $this->ui_factory->table()->column()->text($this->lng->txt('lastname'))
+                                           ->withIsSortable(true),
+        ];
+    }
+
+    protected function getActions(): array
+    {
+        $query_params_namespace = ['frm_moderators_table'];
+
+        $uri_detach = $this->data_factory->uri(
+            ILIAS_HTTP_PATH . '/' . $this->ctrl->getLinkTargetByClass(
+                ilForumModeratorsGUI::class,
+                'detachModeratorRole'
+            )
+        );
+
+        $url_builder_detach = new UI\URLBuilder($uri_detach);
+        list(
+            $url_builder_detach, $action_parameter_token_copy, $row_id_token_detach
+        ) =
+            $url_builder_detach->acquireParameters(
+                $query_params_namespace,
+                'action',
+                'usr_ids'
+            );
+
+        return [
+            'detachModeratorRole' => $this->ui_factory->table()->action()->multi(
+                $this->lng->txt('remove'),
+                $url_builder_detach->withParameter($action_parameter_token_copy, 'detachModeratorRole'),
+                $row_id_token_detach
+            ),
+        ];
+    }
+
+    protected function getDataRetrieval(): UI\Component\Table\DataRetrieval
+    {
+        $data_retrieval = new class($this->forum_moderators) implements UI\Component\Table\DataRetrieval {
```

Comment:
One (hopefully) last thing: Can you please try to check how often this class is created?

If the answer/result ist `> 1`, please don't use an anonymous class for the `DataRetrieval` implementation. Instead, let class `ForumModeratorsTable` implement `\UI\Component\Table\DataRetrieval` and move the mathods as well.